src/app/auth/forgotPassword/index.tsx

```diff
@@ -1,116 +1,101 @@
 import { router } from 'expo-router';
-import React, { useState } from 'react';
-import { Alert, TextInput, View } from 'react-native';
-import { Button, Input } from 'react-native-elements';
+import React, { useState, useRef } from 'react';
+import { Alert, Text, View } from 'react-native';
+import validator from 'validator';
+import { Mutex } from 'async-mutex';
 
 import styles from './styles';
 import globalStyles from '../../../styles/globalStyles';
 import { useSession } from '../../../utils/AuthContext';
+import UserStringInput from '../../../components/UserStringInput/UserStringInput';
+import StyledButton from '../../../components/StyledButton/StyledButton';
+import { isEmailTaken } from '../../../queries/profiles';
+import { queryEmailByUsername } from '../../../queries/auth';
+import colors from '../../../styles/colors';
 
 function ForgotPasswordScreen() {
   const { updateUser, signOut, resetPassword, verifyOtp } = useSession();
   const [email, setEmail] = useState('');
-  const [password, setPassword] = useState('');
-  const [loading, setLoading] = useState(false);
-  const [verificationCode, setCode] = useState<string>('');
-  const [changingPassword, setChangingPassword] = useState(false);
+  const [emailToReset, setEmailToReset] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [validEmail, setValidEmail] = useState(false);
+
+  const mutex = useRef(new Mutex());
 
   const sendResetEmail = async () => {
-    const { error } = await resetPassword(email);
+    const { error } = await resetPassword(emailToReset);
     if (error)
       Alert.alert('Could not send a reset password email. Please try again.');
-    else
-      Alert.alert(
-        'Enter the verification code from your email to change your password',
-      );
+    else router.replace('auth/signup');
   };
-  const verifyCode = async () => {
-    setLoading(true);
 
-    if (email && verificationCode) {
-      const { error } = await verifyOtp(email, verificationCode);
+  const setAndCheckEmailOrUsername = async (newEmail: string) => {
+    setEmail(newEmail);
+    const release = await mutex.current.acquire();
```

Comment:
So I found a much better way to fix this issue (sorry about making you implement the mutex, but this makes the code a lot easier to understand). Using this [library](https://github.com/xnimorz/use-debounce?tab=readme-ov-file) (https://github.com/xnimorz/use-debounce?tab=readme-ov-file), we can debounce a value. This means the state will only update when the user stops updating the text field (aka, stops typing). So we only need to run database requests whenever the user stops typing and not everytime the text box updates. It should be really easy to implement and you can play around with what you want the debounce value to be.